lms/djangoapps/verify_student/api.py

```diff
@@ -33,3 +40,76 @@ def send_approval_email(attempt):
     else:
         email_context = {'user': attempt.user, 'expiration_datetime': expiration_datetime.strftime("%m/%d/%Y")}
         send_verification_approved_email(context=email_context)
+
+
+def create_verification_attempt(user, name, status, expiration_datetime=None):
+    """
+    Create a verification attempt.
+
+    This method is intended to be used by IDV implementation plugins to create VerificationAttempt instances.
+
+    Args:
+        user (User): the user (usually a learner) performing the verification attempt
+        name (string): the name being ID verified
+        status (string): the initial status of the verification attempt
+        expiration_datetime (datetime, optional): When the verification attempt expires. Defaults to None.
+
+    Returns:
+        id (int): The id of the created VerificationAttempt instance
+    """
+    verification_attempt = VerificationAttempt.objects.create(
+        user=user,
+        name=name,
+        status=status,
+        expiration_datetime=expiration_datetime,
+    )
+
+    return verification_attempt.id
+
+
+def update_verification_attempt(attempt_id, name=None, status=None, expiration_datetime=None):
+    """
+    Update a verification attempt.
+
+    This method is intended to be used by IDV implementation plugins to update VerificationAttempt instances.
+
+    Arguments:
+        * attempt_id (int): the verification attempt id of the attempt to update
+        * name (string, optional): the new name being ID verified
+        * status (string, optional): the new status of the verification attempt
+        * expiration_datetime (datetime, optional): The new expiration date and time
+
+    Returns:
+        * None
+    """
+    try:
+        attempt = VerificationAttempt.objects.get(id=attempt_id)
+    except VerificationAttempt.DoesNotExist:
+        log.error(
+            f'VerificationAttempt with id {attempt_id} was not found '
+            f'when updating the attempt to status={status}',
+        )
+        raise
+
+    if name is not None:
+        attempt.name = name
+
+    if status is not None:
+        attempt.status = status
+
+    if expiration_datetime is not None:
+        attempt.expiration_datetime = expiration_datetime
+
+    try:
+        attempt.clean_fields()
+        attempt.save()
+    except ValidationError:
```

Comment:
I had to add this call to `attempt.clean_fields()` in order for this ValidationError to work as @MichaelRoytman mentioned in their previous comment. Let me know if there's something else I should do that's best practice.

Reply:
this reads like it would take _any_ validation error and report back that the status is invalid. If we're only checking for bad status we should do that a bit more explicitly. I'd fear this would create confusing behavior where another field fails to validate but our system incorrectly tells us the status is bad.

Reply:
We should probably just call `full_clean()` here to validate the changes fully if we take this approach.

That said, maybe it would be easier to do what you had before and validate the status values in Python? I didn't realize that calling `attempt.save()` wouldn't perform that validation.

I haven't seen the use of Django forms a lot at edX, but using a form would perform the validation automatically, because calling `save()` on a form will call `full_clean()`.

But I agree with Zach. The validation error will look something like this, so you could key off that.

```
django.core.exceptions.ValidationError: {'status': ["Value 'invalid' is not a valid choice."]}
```

Reply:
I put back the code to only validate which I had earlier